rest/data/volume.go

```diff
@@ -0,0 +1,81 @@
+package data
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/pkg/errors"
+)
+
+var errHostStatusChangeConflict = errors.New("conflicting host status modification is in progress")
+
+// To be moved to a better home when we restructure the resolvers.go file
+// TerminateSpawnHost is a shared utility function to terminate a spawn host
+func TerminateSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
```

Comment:
I recognize that it's unusual to have functions in rest/data that aren't in the connector; however, since we intend to remove the connector, I think functions that are used between service/graphql/etc will still need a neutral home. (This can't live in the host package because of import cycles with unit)

Reply:
it feels strange to me to have spawnHost functions in volume.go since they are logically different. can we maybe create a rest/data/spawnHost.go?